extensions/jdbc/jdbc-h2/deployment/src/main/java/io/quarkus/jdbc/h2/deployment/H2JDBCReflections.java

```diff
@@ -21,4 +38,49 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, driverName));
     }
 
+    @BuildStep
+    NativeImageFeatureBuildItem enableH2Feature() {
+        return new NativeImageFeatureBuildItem("io.quarkus.jdbc.h2.runtime.H2Reflections");
+    }
+
+    /**
+     * We need to index the H2 database core jar so to include custom extension types it's
+     * loading via reflection.
+     */
+    @BuildStep
```

Comment:
Maybe only do this if doing a native build? So as not to slow down JVM/dev-mode pointlessly.

I know it's probably not that much of a hit, but you'll be the first to agree little things tend to pile up.

```suggestion
    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
```

Reply:
You're right, I missed it (had done it on the two methods below)